### PR TITLE
Add @GetMapping, @GetMapping (user_Id) , @DeletMapping , @PostMapping (authorization)

### PR DESCRIPTION
**Pull request by Jundillah123**
Jepretan Layar 2024-12-14 pukul 15 13 35

### CODE REVIEW

**Comment by Jundillah123** [comment]
Jepretan Layar 2024-12-14 pukul 17 36 13
@DeleteMapping

File: src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
 
     private MCifService mCifService;
 
+    @Autowired
     MCifController(MCifService mCifService) {
         this.mCifService = mCifService;
     }
 
+    @GetMapping
+    public List<MCifResponse> readS(@Param("customerName") String customerName) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java]: rename to readMCifList

File: src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.List;
+import java.util.Optional;
+
 @RestController
 @RequestMapping("/request")
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java]: refactor menjadi "/cif"

File: src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
 
+import com.b2camp.simple_core_banking.dto.TSavingAccountResponse;
 import com.b2camp.simple_core_banking.service.TSavingAccountService;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/tsaving")
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java]: refactor to t-saving

File: src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java [hunk 4]
```diff
@@ -13,15 +13,14 @@ public class MCifResponse {
     private String email;
     private String idNumber;
     private String idNumberType;
-    private boolean isDeleted;
-    private LocalDateTime createdAt;
-    private LocalDateTime upDateAt;
+    private Boolean isDeleted;
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java]: refactor type data to primitive

File: src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountRequest.java [hunk 5]
```diff
@@ -1,9 +1,66 @@
 package com.b2camp.simple_core_banking.dto;
 
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+
 public class TSavingAccountRequest {
     private String accountNumber;
     private String cifId;
+    private String mCifId;
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountRequest.java]: rename to cifId

**Review by Yaksyin — APPROVED** [review]

**Review by xuibdg — APPROVED** [review]

**Pull request merged**